### PR TITLE
Support datetime alternatives type

### PR DESCRIPTION
**Pull request by rubio41**
add type DatetimeAlt under CalendarMode.
Support the merge of:
DateTime & Time
Date & Date
Time & Time
DateTimeRange & Time Range
TimeRange & TimeRange

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Definitions/French/DateTimeDefinitions.cs [hunk 1]
```diff
@@ -193,6 +193,7 @@ public static class DateTimeDefinitions
 		public const string GeneralEndingRegex = @"^[.]";
 		public const string MiddlePauseRegex = @"^[.]";
 		public const string DurationConnectorRegex = @"^[.]";
+		public const string OrRegex = @"^[.]";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions/French/DateTimeDefinitions.cs]: Add TODOs for the other languages.

Comment (rubio41) [reply .NET/Microsoft.Recognizers.Definitions/French/DateTimeDefinitions.cs]: the TODOs are all in the yaml file. The definitions class is auto-generated.

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishDateTimeALTExtractorConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+﻿using System.Text.RegularExpressions;
+
+using Microsoft.Recognizers.Definitions.English;
+using System.Collections.Generic;
+
+namespace Microsoft.Recognizers.Text.DateTime.English
+{
+    public class EnglishDateTimeALTExtractorConfiguration : IDateTimeALTExtractorConfiguration
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishDateTimeALTExtractorConfiguration.cs]: Rename ALT to Alt in class names and data types.

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateTimeALTExtractor.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+        private Dictionary<string, object> ExtractALT(ExtractResult former, ExtractResult latter)
+        {
+            var data = ExtractDateTime_Time(former, latter);
+
+            if (data.Count == 0)
+            {
+                data = ExtractDate_Date(former, latter);
+            }
+            if (data.Count == 0)
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateTimeALTExtractor.cs]: Nitpick: blanks between ifs

File: .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimeALTParser.cs [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+                    dateTimePr = this.config.DateTimePeriodParser.Parse(dateTimeEr, referenceTime);
+                }
+            }
+
+            if (dateTimePr.Value != null)
+            {
+                ret.FutureValue = ((DateTimeResolutionResult)dateTimePr.Value).FutureValue;
+                ret.PastValue = ((DateTimeResolutionResult)dateTimePr.Value).PastValue;
+                ret.Timex = dateTimePr.TimexStr;
+                // creat resolution
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimeALTParser.cs]: create

File: .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseMergedParser.cs [hunk 5]
```diff
@@ -454,10 +458,67 @@ internal void ResolveAmpm(Dictionary<string, object> resolutionDic, string keyNa
...
                 AddPeriodToResolution(resolutionDic, TimeTypeConstants.START_DATETIME, TimeTypeConstants.END_DATETIME, mod, res);
             }
+            else if (type.Equals(Constants.SYS_DATETIME_DATETIMEALT))
+            {
+                // for a period
+                if (resolutionDic.Count > 2)
+                {
+                    AddAltPeriodToResolution(resolutionDic, mod, res);
+                }
+                // for a datetime point
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseMergedParser.cs]: Move the comment inside the else

File: Specs/DateTime/English/DateTimeModelExtendedTypes.json [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+            "value": "2017-12-14",
+            "context": "next week"
+          }
+        ]
+      }
+    }
+  ]
+},
+{
+  "Input": "Cortana could try to arrange a Skype call next week on Tuesday 9 am or Thursday 1 pm please?",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModelExtendedTypes.json]: "next week on Tuesday 9 am or Thursday 1 pm" should be tagged as ParentText and be a top-level attr of each result.

File: Specs/DateTime/English/DateTimeModelExtendedTypes.json [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+  "Context": {
+    "ReferenceDateTime": "2017-12-04T00:00:00"
+  },
+  "NotSupported": "javascript",
+  "NotSupportedByDesign": "python",
+  "Results": [
+    {
+      "Text": "next week on tuesday 9 am",
+      "TypeName": "datetimeV2.datetime",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModelExtendedTypes.json]: This should be datetimealt

**Comment by tellarin** [comment]
Closing this as PR https://github.com/Microsoft/Recognizers-Text/pull/267 solves the conflicts.

**Pull request closed without merging**